### ISSUE TITLE
Value of type 'AuthCredential?' has no member 'accessToken'

### ISSUE THREAD

**Issue by gledrich** [firebase/firebase-ios-sdk#5910]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.5
  * Firebase SDK version: 6.27.0
  * Firebase Component: Auth
  * Component version: 6.27.0
  * Installation method: `CocoaPods`

### [REQUIRED] Step 2: Describe the problem

**Reference:** https://firebase.google.com/docs/auth/ios/github-auth#handle_the_sign-in_flow_with_the_firebase_sdk

#### Steps to reproduce:

Similarly to this closed issue https://github.com/firebase/firebase-ios-sdk/issues/3076, I am unable to access members of AuthCredential as described here https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIROAuthCredential - in particular the accessToken. The reference suggests that the

```
          // GitHub OAuth access token can also be retrieved by:
          // authResult.credential.accessToken
```

however this property is not available due to the error
`Value of type 'AuthCredential?' has no member 'accessToken'`

#### Relevant Code:

```
provider.getCredentialWith(nil) { credential, error in
      if error != nil {
        // Handle error.
      }
      if credential != nil {
        Auth.auth().signIn(with: credential) { authResult, error in
          if error != nil {
            // Handle error.
          }
          // User is signed in.
          // IdP data available in authResult.additionalUserInfo.profile.
          // GitHub OAuth access token can also be retrieved by:
          // authResult.credential.accessToken
          // GitHub OAuth ID token can be retrieved by calling:
          // authResult.credential.idToken
        }
      }
    }
```

**Notes**
* I have tried this with no luck https://github.com/firebase/firebase-ios-sdk/issues/3076#issuecomment-506282939
* `Auth().signIn(with: credential) { authResult, error in` in the documentation should be corrected to `Auth.auth().signIn(with: credential) { authResult, error in`

**Comment by scottcrossen** [comment]
This looks like a typing error. Are you able to cast [AuthCredential](https://firebase.google.com/docs/reference/swift/firebaseauth/api/reference/Classes/AuthCredential) to [OauthCredential](https://firebase.google.com/docs/reference/swift/firebaseauth/api/reference/Classes/OAuthCredential) ?

**Comment by gledrich** [comment]
@scottcrossen I'm new to swift so bare with me... The following

`let cred:OAuthCredential = authResult?.credential`

throws this error

`Cannot convert value of type 'AuthCredential?' to specified type 'OAuthCredential'`

**Comment by gledrich** [comment]
I hacked something together and can confirm the access token is available when down casted to OAuthCredential

**Comment by morganchen12** [comment]
Docs fixed in cl/336168903 (Googler-only link)
